### ISSUE TITLE
cif_core.dic: conflicting _valence_param category definitions

### ISSUE THREAD

**Issue by vaitkus** [COMCIFS/cif_core#62]
**Edit:** the original issue has been resolved by merging PR #568. This issue remains open as a reminder to potentially mention a few additional points about the valence categories in the upcoming version of Volume G (see https://github.com/COMCIFS/cif_core/issues/62#issuecomment-3508396092).

Dictionary translation from DDL1 to DDLm seems to have changed the way data items in the ``_valence_param`` are interpreted. The following changes were made:
1) Data items ``_valence_param.atom_1_valence`` and ``_valence_param.atom_2_valence`` were assigned a range limit of ``[0.0; inf]``.
2) Data items ``_valence_param.atom_1_valence`` and ``_valence_param.atom_2_valence`` were declared as being measured in coulombs;
3) Data item ``_valence_param_B`` was asigned a range limit of `[0.1:]`;
4) Data item ``_valence_param_Ro`` was asigned a range limit of `[1.0:]`;

This might seem correct at first glance, but after an inspection of existing CIFs it seems that it conflicts greatly with the way these data items were used in ``DDL1``. The following CIF snippet (taken from
http://www.iucr.org/resources/data/data-sets/bond-valence-parameters) provides a good example of how it was used:

```
loop_
   _valence_param_atom_1
   _valence_param_atom_1_valence
   _valence_param_atom_2
   _valence_param_atom_2_valence
   _valence_param_Ro
   _valence_param_B
   _valence_param_ref_id
   _valence_param_details
Ac 3   O  -2    2.24     0.37     b   ?
Ac 3   O  -2    2.29     0.35     p   ?
Ac 3   F  -1    2.13     0.37     b   ?
Ac 3   F  -1    2.10     0.40     p   ?
...
H  1   O  -2    0.569    0.94     bc  '1.05<O-H<1.70 A, best general value'
H  1   O  -2    0.907    0.28     bc  'O-H < 1.05 A'
H  1   O  -2    0.990    0.59     bc  '1.70 A < O-H'
H  1   O  -2    0.918    0.427    bs  ?
H  1   O  -2    0.925    0.40     az  ?
```
The following conclusions can be made from the example:
a) The range specified in 1) would be correct for the actual valences of the atoms, but from the definition of these data items ("``The valence (formal charge) of the atom  whose bond valence parameters are given in this category.``") and the example it seems that the data names are actually misnomers. If the author of the definitions indeed had formal charges and not valencies in mind negative values should not be forbidden;
b) If indeed the data names ``_valence_param.atom_?_valence`` provide the formal charge of the atom, measuring them in coulombs is very inconvenient. Maybe an additional unit code 'elementary charge' could be defined? It would probably serve well in the definitions of other data items as well. The example showcases elementary charges already being used;
c) Currently the ``_valence_param.atom_?_valence`` is defined as real number. I'm not sure what the IUCr policy is on specifying partial charges, but it should probably be modified (or not) in accordance to it;
d) I am not sure what limitations the valence-bond model imposes on the **R<sub>0</sub>** and **B** parameters, but the given example clearly showcases ``_valence_param_Ro`` values going below 1. Maybe they should be restricted just to positive values to play it safe?

**Comment by jamesrhester** [comment]
I agree with these comments. I wil adjust the definitions accordingly. Note that units of 'electrons' are available for use here.

**Comment by vaitkus** [comment]
Thank You for the corrections.

**Comment by jamesrhester** [comment]
Units of 'electrons' are not quite right, as '-2' would mean -2 electrons, or a formal charge of +2.  I think we can safely say that it is dimensionless until further notice.

**Comment by jamesrhester** [comment]
This looks generally pretty good. Instead of `valence_ref`, could we instead get `_valence_param.ref_id` to point to an item in the more general `citation` list, deprecating `valence_ref`? I note that these tags appear not be present in COD, so perhaps we can just retask `_valence_param.ref_id` instead of defining a new `_valence_param.citation_id`.

**Comment by vaitkus** [comment]
I have glanced at the IUCr website once more and have two additional notes:
- The website states that `Bond valence parameters for bonds formed by atoms whose oxidation state is given as 9 do not have an oxidation state specified in the original paper. They may apply to a particular, but unspecified, oxidation state or they may be intended to apply to all oxidation states.`. This notations seems to be used in the data files themselves. It might make sense to indicate this convention in the definitions of the corresponding data items (`_valence_param.atom_?_valence`), although, using '?' or '.' instead of '9' seems more appropriate. I do not known if the authors of that tabulation are open to changes.
- The website also states that value 'NH' in the `_valence_param.atom_?` data item indicates an ammonium ion therefore the existing list already does not correspond only to the chemical element list.

**Comment by vaitkus** [comment]
PR #568 resolved the DDL1 incompatibility.

In the discussion of the same PR, it was suggested that it may be worth mentioning some of the following points in the commentary chapter of volume G:
- There is a tabulation of bond-valence parameters available on the IUCr website ([1]).
- Older versions of this tabulation used a convention where the valence of '9' should be interpreted as an unspecified valence (`... do not have an oxidation state specified in the original paper.`) 

[1] https://www.iucr.org/resources/data/datasets/bond-valence-parameters
